### DISCUSSION TITLE
File Downloader

### DISCUSSION THREAD

**Discussion by SunshineSpring666** [CommunityToolkit/Maui#554] [New Feature Discussions]
MAUI is awesome. Some apps need basic functionalities like downloading files (documents, images, videos, etc.) or printing files (documents, images, etc.). Existing libraries for Xamarin raises puzzling exceptions when migrated to MAUI projects. We spend days trying to solve these, but failed. 
    I wonder if you guys have such basic component requirements in MAUI apps. To my knowledge, there are no such libs for MAUI now. Any clues on how to get "file download" and "printing by printer" functionalities in MAUI (on Windows/Android/iOS)?
    Thanks.

**Comment by VladislavAntonyuk** [comment]
You can use WebClient to download files.
It is available in .NET so you don’t need to use.NET MAUI

**Comment by pictos** [comment]
@SunshineSpring666 I did [this code](https://github.com/pictos/DownloadManager/blob/master/DownloadManager/DownloadManager.shared.cs) while ago, with it you will be able to download files using httpclient

**Comment by SunshineSpring666** [comment]
Thank Vladislav and Pedro for your reply. You guys are right. WebClient can easily let people handle percentage info, while HttpClient is the officially recommended one. Thank you all! I'd move the MAUI printing discussion to a separate thread for simplicity.

**Comment by CliffAgius** [comment]
Closed as answered
